discord/client.py

```diff
@@ -4947,3 +4950,45 @@ async def report_unverified_application(
             icon_data = utils._bytes_to_base64_data(icon.fp.read())
             await state.http.upload_unverified_application_icon(app.name, app.hash, icon_data)
         return app
+
+    @overload
+    async def fetch_experiments(self, with_guild_experiments: Literal[True] = ...) -> ExperimentResponseWithGuild:
+        ...
+
+    @overload
+    async def fetch_experiments(self, with_guild_experiments: Literal[False] = ...) -> ExperimentResponse:
+        ...
+
+    async def fetch_experiments(
+        self, with_guild_experiments: bool = True
+    ) -> Union[ExperimentResponse, ExperimentResponseWithGuild]:
+        """|coro|
+
+        Retrieves the experiment assignments for the user.
+
+        Parameters
+        -----------
+        with_guild_experiments: :class:`bool`
+            Whether to include guild experiment rollouts in the response.
+
+        Raises
+        -------
+        HTTPException
+            Retrieving the experiment assignments failed.
+
+        Returns
+        -------
+        List[Union[:class:`.UserExperiment`, :class:`.GuildExperiment`]]
+        """
+        state = self._connection
+        data = await state.http.get_experiments(with_guild_experiments=with_guild_experiments)
+
+        return data
```

Comment:
why are you returning raw json when you have models

Reply:
I thought for direct HTTP calls I should return the API values but I guess in the client function I should’ve deserialised